### PR TITLE
TINY-9965: Deleting content within the <details> of a nested accordion, removes nested accordion <summary> content

### PR DESCRIPTION
**Pull request by spocke**
Related Ticket: TINY-9965

Description of Changes:
* Fixed issue with deleting blocks before/after nested details elements.
* Added more low level tests for details delete code.
* Separated the Safari specific delete in summary code form the rest of the details delete code.
* Various code cleanup.

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 - Returning an empty string in custom context menu update function would result in a small white line appearing on right click and the native browser context menu would not show up. #TINY-9842
 - Tab navigation incorrectly stopped around `iframe` dialog component. #TINY-9815
+- It was possible to delete the last empty block just before a details element if it was nested within another details element. #TINY-9965
```
Comment (vpyshnenko) [inline modules/tinymce/CHANGELOG.md:16]:
```suggestion
- It was possible to delete the sole empty block just before a details element if it was nested within another details element. #TINY-9965
```

**Review by vpyshnenko — APPROVED** [review]

File: modules/tinymce/CHANGELOG.md (hunk 1 above)
Comment (spocke) [reply modules/tinymce/CHANGELOG.md]: Yes, was struggling to describe that in a changelog item. Thanks.

File: modules/tinymce/src/core/main/ts/selection/DetailsElement.ts [hunk 2]
```diff
@@ -149,23 +151,20 @@ const preventDeleteIntoDetails = (editor: Editor, forward: boolean) => {
...
-          });
-          return true;
-        }
-      } else if (Type.isNull(parentBlock.previousSibling)) {
-        const pos = CaretFinder.nextPosition(root, caretPos).filter((pos) => isInDetailsElement(dom, pos));
-        if (pos) {
+    const parentDetailsAtCaret = getParentDetailsElementAtPos(dom, caretPos);
+    const inEmptyParentBlock = parentBlock && dom.isEmpty(parentBlock);
+    const isFirstBlock = parentBlock && Type.isNull(parentBlock.previousSibling);
+    const isLastBlock = parentBlock && Type.isNull(parentBlock?.nextSibling);
```
Comment (lorenzo-pomili) [inline modules/tinymce/src/core/main/ts/selection/DetailsElement.ts:157]: why is one row you use `parentBlock?` and in the row above only `parentBlock` without the `?`?

**Review by lorenzo-pomili — APPROVED** [review]

File: modules/tinymce/src/core/main/ts/selection/DetailsElement.ts (hunk 2 above)
Comment (spocke) [reply modules/tinymce/src/core/main/ts/selection/DetailsElement.ts]: Right I changed both to be null coalescing. I'm still getting used to use that feature of the language. :smile:

**Pull request merged**